xclim/core/indicator.py

```diff
@@ -454,157 +459,77 @@ def from_dict(
           The module name of the indicator. This is meant to be used only if the indicator
           is part of a dynamically generated submodule, to override the module of the base class.
         """
-        # Make cell methods. YAML will generate a list-of-dict structure, put it back in a space-divided string
-        if data.get("output", {}).get("cell_methods") is not None:
-            cell_methods = parse_cell_methods(data["output"]["cell_methods"])
-        else:
-            cell_methods = None
+        data = data.copy()
+
+        if "base" in data:
+            if isinstance(data["base"], str):
+                cls = registry.get(
+                    data["base"].upper(), base_registry.get(data["base"])
+                )
+                if cls is None:
+                    raise ValueError(
+                        f"Requested base class {data['base']} is neither in the indicators registry nor in base classes registry."
+                    )
+            else:
+                cls = data["base"]
+        elif cls is Indicator:
+            raise ValueError(
+                "Indicators can't be created from dict with the Indicator class as base. Please use an existing indicator as base, or another base class."
+            )
```

Comment:
This is born from a need to use `from_dict` outside a yaml file, but without knowing before hand which class shoud be subclassed. It's strange to change the `cls` var, but I haven't found another clean way...